scripts/generate_rust_analyzer.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
+srctree=$1
+objtree=$2
+lib_src=$3
+bindgen_file=$4
+
+
+# TODO: create issue at RA to accept @arg file
+function generate_cfgs() {
+    echo '['
+    cfgs=$(cat $objtree/include/generated/rustc_cfg | sed 's/--cfg=//' | sed 's/"/\\"/g' | sed 's/$/",/' | sed 's/^/"/')
+    echo ${cfgs::-1}
+    echo ']'
+}
+
+function generate_crate() {
+    name=$1
+    module=$2
+    member=${3:-"true"}
+    cfg=${4:-$(generate_cfgs)}
+    deps=${5:-'[{"crate":0,"name":"core"},{"crate":2,"name":"alloc"},{"crate":4,"name":"kernel"}]'}
+    extra=${CRATE_EXTRA:-""}
+
+    echo "{
+        \"display_name\":\"$name\",
+        \"root_module\":\"$module\",
+        \"edition\":\"2018\",
+        \"deps\":$deps,
+        \"is_workspace_member\": $member,
+        \"cfg\": $cfg,
+        \"env\": {
+            \"RUST_BINDGEN_FILE\": \"$bindgen_file\",
+            \"RUST_MODFILE\": \"compile_error!(\\\"This is only for rust-analyzer\\\");\"
```

Comment:
You'll need to change the `RUST_MODFILE` value to something else since this gets [transformed](https://github.com/Rust-for-Linux/linux/blob/912beef787b400b0518722c15879136c65b18608/rust/module.rs#L680) to
```Rust
#[cfg(not(MODULE))]
#[link_section = ".modinfo"]
#[used]
pub static __rust_random_file: [u8; 67] = *b"rust_random.file=compile_error!("This is only for rust-analyzer");\0";
```
which is invalid (notice the `"`) and causes rust-analyzer to choke:
![2021-04-29_20-29-16_grim](https://user-images.githubusercontent.com/2302947/116600491-a613c980-a929-11eb-95fe-6332b2adc585.png)
Options are "upgrading" from `\\\` to `\\\\\\\` (+4 since we're escaping through three levels...) or probably better to just use `"RUST_MODFILE": "This is only for rust-analyzer"` which should then greet you with a diagnostic:
![2021-04-29_20-34-20_grim](https://user-images.githubusercontent.com/2302947/116601079-54b80a00-a92a-11eb-8c44-0d76577d8ae5.png)

Reply:
Oh, missed that. It also does not show that on my side. Not sure why I don't get that error.

Reply:
Hm, there's a setting called `rust-analyzer.procMacro.enable` but it should already be enabled by default...

Reply:
I can see the values generated by the procMarco. But I don't see warnings from the Marco itself.

Reply:
That was my bad. I've picked up the [Error Lens](https://marketplace.visualstudio.com/items?itemName=usernamehw.errorlens) VS Code extension from one of the rust-analyzer changelog which renders the diagnostics like that. You should still be able to see the messages by hovering over `module!` and there should also be a small yellow dashed line below the `m` as in my screenshots above.